membership, policy: add SIG, WG and subproject roles

dhiller opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

**What this PR does / why we need it**:

As a followup to #301 this PR adds roles:
* SIG chair
* SIG Subproject Lead
* WG chair

Fixes a couple of header size issues in existing parts along the way.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note

```

dhiller commented:
@aburdenthehand WDYT? Would that be enough as a first followup?

aburdenthehand commented on membership_policy.md:
```diff
@@ -2,13 +2,15 @@
+| Contributor                                 | Adhere to the [KubeVirt code of conduct](https://github.com/kubevirt/community/blob/main/code-of-conduct.md) | Submit at least one pull request                                                                                    | Active GitHub account                |
+| Member                                      | Active contributor in the community                                                                          | * Active contributor<br>* Sponsored by 2 members<br>* Multiple contributions to the project                         | KubeVirt GitHub org member           |
+| Reviewer                                    | Review contributions from others                                                                             | * Member<br>* History of review and authorship in the project<br> * Sponsored by an approver                        | [OWNERS_ALIASES] file reviewer entry |
+| Approver                                    | Approve accepting contributions                                                                              | * Reviewer<br>* Highly experienced<br> * Active reviewer & contributor to the project<br>Sponsored by 2 approvers   | [OWNERS_ALIASES] file approver entry |
```
```suggestion
| Approver                                    | Approve accepting contributions                                                                              | * Reviewer<br>* Highly experienced<br> * Active reviewer & contributor to the project<br> * Sponsored by 2 approvers   | [OWNERS_ALIASES] file approver entry |
```

aburdenthehand commented on membership_policy.md:
```diff
@@ -128,6 +130,45 @@ The following apply to the part of project for which one would be an approver in
   * Mentor contributors and reviewers
   * May approve contributions for acceptance
 
+## SIG Chair
```
```suggestion
## Special Interest Group (SIG) Chair
```

aburdenthehand commented on membership_policy.md:
```diff
@@ -128,6 +130,45 @@ The following apply to the part of project for which one would be an approver in
+* is an active reviewer
+* may claim approver rights in scope
+
+## WG Chair
```
```suggestion
## Working Group (WG) Chair
```

aburdenthehand commented on membership_policy.md:
```diff
@@ -128,6 +130,45 @@ The following apply to the part of project for which one would be an approver in
+
+### Responsibilities and privileges
+
+* is an organizer and facilitator
```
of what?

aburdenthehand commented on membership_policy.md:
```diff
@@ -128,6 +130,45 @@ The following apply to the part of project for which one would be an approver in
+
+### Requirements
+
+* contributor
```
I'd probably want SIG chairs to have been 'Members for at least 3 months' , similar to what we have for 'Reviewer'. Maybe even better to have been 'Reviewer for at least 3 months'? Will there be a scenario where that would be an issue?

dhiller replied:
Suggestion: sig-reviewer for at least 3 months

dhiller replied:
resorted to another change

aburdenthehand commented on membership_policy.md:
```diff
@@ -128,6 +130,45 @@ The following apply to the part of project for which one would be an approver in
+
+* leads within subproject scope
+* is an active reviewer
+* may claim approver rights in scope
```
Not sure I understand what this means

dhiller replied:
elaborated on that

aburdenthehand commented on membership_policy.md:
```diff
@@ -128,6 +130,45 @@ The following apply to the part of project for which one would be an approver in
+
+### Requirements
+
+* contributor
```
Member, as above

aburdenthehand commented on membership_policy.md:
```diff
@@ -128,6 +130,45 @@ The following apply to the part of project for which one would be an approver in
+
+### Requirements
+
+* contributor
```
Member, as above

aburdenthehand commented on membership_policy.md:
```diff
@@ -128,6 +130,45 @@ The following apply to the part of project for which one would be an approver in
+### Responsibilities and privileges
+
+* is an organizer and facilitator
+* gives updates to respective sponsoring SIG Chairs
```
I'd want all these points to start with upper case ;)

dhiller replied:
BUT WHY?

dhiller replied:
Kidding - can do

aburdenthehand reviewed (commented):
@dhiller My gosh you're quick. Thanks for putting this together. Yeah, I think this is a great start. The table seems pretty solid, however I can see some more things to add or change in the descriptions - probably already known since this is a draft.

dhiller commented:
> @dhiller My gosh you're quick. Thanks for putting this together. Yeah, I think this is a great start. The table seems pretty solid, however I can see some more things to add or change in the descriptions - probably already known since this is a draft.

Hey @aburdenthehand thanks for your review - I've elaborated on the roles a bit, hope that works better now.

Along the way I've fixed some issues with markdown header levels in the remaining document, and I've added some links so that you can get to the role faster.

PTAL, thank you for your time!

xpivarc commented:
/cc @davidvossel @alicefr @jean-edouard @mhenriks @EdDev @enp0s3 @acardace

xpivarc commented:
/cc

aburdenthehand reviewed (commented):
I've suggested summaries for these new roles, similar to what we have for the established roles. 
I also found a couple of nits.

dhiller commented:
> I've suggested summaries for these new roles, similar to what we have for the established roles. I also found a couple of nits.

Thank you @aburdenthehand for the review, I've added the suggestions, PTAL! :pray:

aburdenthehand commented:
/lgtm

EdDev commented on membership_policy.md:
```diff
@@ -2,13 +2,15 @@
+| [Member](#member)                              | Active contributor in the community                                                                          | * Active contributor<br>* Sponsored by 2 members<br>* Multiple contributions to the project                          | KubeVirt GitHub org member           |
+| [Reviewer](#reviewer)                          | Review contributions from others                                                                             | * Member<br>* History of review and authorship in the project<br> * Sponsored by an approver                         | [OWNERS_ALIASES] file reviewer entry |
+| [Approver](#approver)                          | Approve accepting contributions                                                                              | * Reviewer<br>* Highly experienced<br> * Active reviewer & contributor to the project<br> * Sponsored by 2 approvers | [OWNERS_ALIASES] file approver entry |
+| [SIG Chair](#special-interest-group-sig-chair) | Lead a SIG aligned to the goals of the SIG charter                                                           | * sig-approver<br>* Highly experienced in SIG matters<br> * Active reviewer & contributor to the project             | [sigs.yaml] chair entry              |
```
In the section for the SIG chair, the requirement does not mention `sig-approver` but here it is.
And I think we checked this with Kubernetes requirements, which does not mention such a requirement from a chair.

dhiller replied:
@EdDev I have just noticed the paragraph below https://github.com/kubevirt/community/pull/306/files#diff-78a4ac3dbf91e4f4ac17004bc8e0bca8009fcd12e09d46da4b9ab11848e4089aR157

And I think being an approver for the sig as a sig chair does make sense. WDYT?

vladikr replied:
It does to me.

dhiller replied:
> > Claims approver rights in scope
> 
> I read that point based on its content sub-points. I understood it as being responsible to have something like that in a project, not necessarily to also be the one who approves the changes.
> 
> OK, so the suggestion is to couple the chair [1] and tech-lead [2] roles together. Right? The added value of differentiating them is that the chair can administratively manage a SIG while the lead can just take technical decisions (e.g. approve design proposals).

@EdDev I agree that this would be better. I intendedly was a bit more vague, since I am not sure whether we can cover two distinct roles right now, given the number of contributors that we have currently that can and may want to fill such role(s).

Maybe it makes sense to combine them for now and if we see the need to separate the responsibilities we do that later? As of today I have the impression that qualified contributors would be able to fill both roles anyway.

Also there's still the possibility of delegation to the subproject leads. WDYT?

> 
> We already had one incident like this, when we had a volunteer who agreed to lead a SIG but was not yet a maintainer/approver. I do not intent to insist over this, but please be aware of the cost of unifying the two. We already have an issue with approvers managing to invest enough time in the technical area, adding more responsibility over it without an alternative will make their work harder.
> 
> [1] [kubernetes/community@`master`/committee-steering/governance/sig-governance.md#chair](https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md?rgh-link-date=2024-09-24T11%3A29%3A36Z#chair) [2] [kubernetes/community@`master`/committee-steering/governance/sig-governance.md#tech-lead](https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md?rgh-link-date=2024-09-24T11%3A29%3A36Z#tech-lead)

EdDev replied:
I am a bit worried that we filter here members that can act as SIG chairs and are not approvers in the repos.

But if you accept this side-effect, we can proceed. If needed, we can reconsider it in the future with a real candidate.

EdDev commented on membership_policy.md:
```diff
@@ -128,6 +130,82 @@ The following apply to the part of project for which one would be an approver in
+* Track record of contributions to SIG matters
+* Advanced knowledge in SIG subject
+
+### Responsibilities and privileges
```
What about its responsibility over the SIG subprojects?

dhiller replied:
So you are asking about making the hierarchy mentioned explicitly? I thought that was already clear?

EdDev replied:
A SIG TL has the responsibility to create subprojects [1].
As we do not define a TL, I guess it is embedded into the chair role.

> MUST Approve & facilitate the creation of new subprojects

[1]  https://github.com/kubernetes/community/blob/master/committee-steering/governance/sig-governance.md#tech-lead

dhiller replied:
My intent here was that the subproject lead is responsible for creating and maintaining the subproject.

dhiller replied:
^^ @EdDev

EdDev replied:
@dhiller , I would just add the quoted point to the list.

> Approve & facilitate the creation of new subprojects

dhiller replied:
Yeah, really good point - I'll update the PR. Thanks!

dhiller commented:
This PR is waiting for three weeks now.

Hey @cwilkers @davidvossel @fabiand @rmohr @vladikr :wave: 

Since @aburdenthehand is out for a while, is one of you able to take a quick look and approve this?

vladikr commented on membership_policy.md:
```diff
@@ -128,6 +130,82 @@ The following apply to the part of project for which one would be an approver in
+
+### Requirements
+
+* Active reviewer for SIG matters for at least 3 months
```
This requirement is a bit problematic for new SIGs...

EdDev replied:
New SIGs will usually be created from existing areas, therefore this can still stand.
In case a new SIG is created from a totally new subject, then the candidate can show knowledge in that are from other sources. But I think this is rarely the case and we could ignore it until such an incident really happens.

vladikr replied:
Yes, that's fine.

EdDev reviewed (commented):
Thank you!

There are only two points that I consider open: SIG chairs required to be approvers and clarifying that SIG chairs are responsible to manage subprojects.

None of them are blockers.

dhiller commented:
@EdDev I have updated the PR and addressed your comments with this commit: https://github.com/kubevirt/community/pull/306/commits/3eff9bad87f71725e30563c8ee884427123cdf30

PTAL, thank you!

vladikr commented:
/approve
Thanks @dhiller !
